### PR TITLE
Protect custom type testers against None values

### LINKED ISSUE #65: CSV file w/ double carriage return break the SIREN/SIRET detector
Not sure if this a bug since the file might not be considered valid in the first place.

Error: 

```
Error: Error parsing CSV: Can not parse value "None" as a SIREN or SIRET. Error at row 0 column epci2020.
```

### PR DESCRIPTION
**Pull request by abulte**
Cf https://github.com/wireservice/agate/blob/master/agate/data_types/boolean.py#L45 for inspiration

Fix #65

### CODE REVIEW

File: csvapi/type_tester.py [hunk 1]
```diff
@@ -25,6 +25,8 @@ def __init__(self, **kwargs):
         super(Time, self).__init__(**kwargs)
 
     def cast(self, d):
+        if d is None:
+            return d
```
Comment (AntoineAugusti) [inline csvapi/type_tester.py:29]: Is this needed? The next line casts to a string before trying to match the regex.

File: csvapi/type_tester.py [hunk 2]
```diff
@@ -36,6 +38,8 @@ def __init__(self):
         super(SirenSiret, self).__init__()
 
     def cast(self, d):
+        if d is None:
+            return d
         if is_valid_siret(d) or is_valid_siren(d):
```
Comment (AntoineAugusti) [inline csvapi/type_tester.py:43]: These functions don't crash for non-string arguments?

**Review by AntoineAugusti — COMMENTED** [review]
Good work, sorry I miss the `None` case for SIREN/SIRET! 👏  for the tests

File: csvapi/type_tester.py (hunk 1 above)
Comment (abulte) [reply csvapi/type_tester.py:29]: tests fail w/o this so I guess so :-)

File: csvapi/type_tester.py (hunk 2 above)
Comment (abulte) [reply csvapi/type_tester.py:43]: I think the tests cover this. And I'd say every value come as a string since it's a CSV and we're trying to guess types.

**Pull request merged**